flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/utils/RelTreeWriterImpl.scala

```diff
@@ -189,7 +191,16 @@ class RelTreeWriterImpl(
         case value =>
           if (j == 0) s.append("(") else s.append(", ")
           j = j + 1
-          s.append(value.left).append("=[").append(value.right).append("]")
+          val rightStr = value.right match {
+            case subQuery: RexSubQuery =>
+              if (withRicherDetailInSubQuery) {
+                computeSubQueryRicherDigest(subQuery)
+              } else {
+                value.right.toString
+              }
+            case _ => value.right.toString
+          }
+          s.append(value.left).append("=[").append(rightStr).append("]")
```

Comment:
Is this block needed for the fixing the table alias clearance?

Reply:
Please see the another answer.